controllers/operator/clusterpodplacementconfig_controller_test.go

```diff
@@ -580,12 +580,73 @@ var _ = Describe("Controllers/ClusterPodPlacementConfig/ClusterPodPlacementConfi
 				validateReconcile()
 			})
 		})
-		AfterAll(func() {
-			err := k8sClient.Delete(ctx, builder.NewClusterPodPlacementConfig().WithName(common.SingletonResourceObjectName).Build())
-			Expect(err).NotTo(HaveOccurred(), "failed to delete ClusterPodPlacementConfig", err)
-			Eventually(framework.ValidateDeletion(k8sClient, ctx)).Should(Succeed(), "the ClusterPodPlacementConfig should be deleted")
+	})
+	When("Creating a cluster pod placement config", func() {
+		Context("with invalid values in the plugins.nodeAffinityScoring stanza", func() {
+			DescribeTable("Should fail creating a ClusterPodPlacementConfig with invalid values in plugins.nodeAffinityScoring", func(object *v1beta1.ClusterPodPlacementConfig) {
+				By("Ensure no ClusterPodPlacementConfig exists")
+				cppc := &v1beta1.ClusterPodPlacementConfig{}
+				err := k8sClient.Get(ctx, crclient.ObjectKey{
+					Name: common.SingletonResourceObjectName,
+				}, cppc)
+				Expect(err == nil || errors.IsNotFound(err)).To(BeTrue(), "The ClusterPodPlacementConfig should not exist")
```

Comment:
```suggestion
				Expect(errors.IsNotFound(err)).To(BeTrue(), "The ClusterPodPlacementConfig should not exist")
```

`err == nil` here would mean that the ClusterPodPlacementConfig was found, hence we should actually fail

Reply:
Done , it's because of the plugin struct it was returning nil and plugin in YAML with empty .